src/components/10-atoms/radio/index.js

```diff
@@ -0,0 +1,251 @@
+  get checked() {
+    const {
+      state: { isControlled, checked, native },
+    } = this;
+    return isControlled ? checked : native;
+  }
+
+  // event handlers
+  handleChange(event) {
+    const {
+      state: { isControlled, checked },
+      name,
+      onChange,
+    } = this;
+    onChange(event);
+
+    const isChecked = event.target.checked;
+
+    // uncontrolled mode:
+    if (!isControlled) {
+      // deselect last selected radiobutton...
+      if (isChecked && selectedRadioButton[name]) {
+        selectedRadioButton[name].checked = false; // causes re-render
+      }
+      // ... and select ourselves
+      this.checked = true; // causes re-render
+
+      return;
+    }
+    // controlled mode:
+    // set UI from model state
+    this.input.checked = checked;
+    // get selected radio button - usually not ourselves
+    const groupSelectedRadioButton = selectedRadioButton[name];
+    // we are already registered in radio-button group?
+    if (name && groupSelectedRadioButton) {
+      // yes, re-select the selected radio button natively, thereby preserving radio behaviour
+      groupSelectedRadioButton.querySelector('input').checked = true;
+    }
+  }
+
+  constructor() {
+    super();
+    // initialize model state
+    this.state = {
+      isControlled: false,
+      firstUpdate: true,
+      checked: false,
+      native: false,
+    };
+    // initialize properties
+    this.value = '';
+    this.name = '';
+    this.label = '';
+    this.icon = '';
+    this.error = '';
+    this.disabled = false;
+    this.button = false;
+    this.nogap = false;
+    this.isReact = false;
+    this.onFocus = () => {};
+    this.onBlur = () => {};
+    this.onChange = () => {};
+  }
+
+  getDefaultName() {
+    this.name = this.name || defaultName();
+    return this.name;
+  }
+
+  render() {
+    const {
+      id,
+      name = this.getDefaultName(),
+      button,
+      icon,
+      checked,
+      value,
+      disabled,
+      label,
+      isReact,
+      state: { isControlled },
+    } = this;
+
+    // controlledness is a React-only notion
+    this.state.isControlled = isControlled && isReact;
+
+    return html`
+      <label class="a-radio__wrapper">
+        <input
+          id="${id}"
+          class="a-radio__input"
+          type="radio"
+          name="${name}"
+          ?checked="${checked}"
+          value="${value}"
+          ?disabled="${disabled}"
+          @change="${this.handleChange}"
+          @focus="${this.onFocus}"
+          @blur="${this.onBlur}"
+        />
+        ${button
+          ? html``
+          : html`
+              <span class="a-radio__icon"></span>
+            `}
+        ${icon ? svg([icon]) : html``}
+        <div class="a-radio__content">${label}</div>
+      </label>
+    `;
+  }
+
+  firstUpdated() {
+    this.input = this.querySelector('input');
+    const { name, button, noautowidth } = this;
+    const ourButton = this.querySelector('.a-radio__content');
```

Comment:
Please dont use dom selection on classes that are used for styling. for this we have a prefix rule: `.js-radio__content` instead of `a-radio__content`. Reason: separation of concern

Reply:
i think we never documented it here but iut was on v1: https://github.com/axa-ch/patterns-library/blob/develop-v1/CONTRIBUTING.md#css

Reply:
Done.

Reply:
@markus-walther seems this is not fixed yet, correct?